app/models/spree/address.rb

```diff
@@ -13,7 +13,7 @@ class Address < ApplicationRecord
     belongs_to :state, class_name: "Spree::State"
 
     has_one :enterprise, dependent: :restrict_with_exception
-    has_many :shipments
+    has_many :shipments, dependent: :nullify
```

Comment:
I think that this is an invalid case. You need to assign a new address to a shipment before you can delete the old one. I'm But usually we update addresses in place and keep the id. I wonder sometimes how many orphan addresses we have because that model is re-used in so many places. Too many places, actually, but that is going off topic here.

Hm, the database and Rails actually allow a shipment without address. Not sure if it causes server problems down the line though... anyway, I guess we never do this.

Reply:
I'll revert this and mark it as TODO for now. It seems like there should be a better solution for this though. For example, we could soft-delete the address to keep it. Rewriting the address seems like we are losing something.